src/main/java/io/snowdrop/github/reporting/GithubReporting.java

```diff
@@ -0,0 +1,307 @@
+  private final ZonedDateTime endTime;
+
+  //These represent the oldest reporting period possible
+  private final Date minStartTime;
+  private final Date minEndTime;
+
+  private final Map<String, Set<Repository>> repositories = new HashMap<>();
+
+  private final Map<String, Set<PullRequestDTO>> openPullRequests = new HashMap<>();
+  private final Map<String, Set<PullRequestDTO>> closedPullRequests = new HashMap<>();
+  private final Map<String, Set<PullRequestDTO>> pullRequests = new HashMap<>();
+
+  private final Map<String, Set<IssueDTO>> openIssues = new HashMap<>();
+  private final Map<String, Set<IssueDTO>> closedIssues = new HashMap<>();
+  private final Map<String, Set<IssueDTO>> issues = new HashMap<>();
+
+  public GithubReporting(RepositoryService repositoryService, PullRequestService pullRequestService,
+      IssueService issueService, int reportingDay, int reportingHour, Set<String> users, Set<String> organizations) {
+    this.repositoryService = repositoryService;
+    this.pullRequestService = pullRequestService;
+    this.issueService = issueService;
+    this.reportingDay = reportingDay;
+    this.reportingHour = reportingHour;
+    this.users = users;
+    this.organizations = organizations;
+
+    this.endTime = ZonedDateTime.now().with(DayOfWeek.of(reportingDay)).withHour(reportingHour);
+    this.startTime = endTime.minusWeeks(1);
+
+    this.minStartTime = Date.from(startTime.minusMonths(1).toInstant());
+    this.minEndTime = Date.from(endTime.minusMonths(1).toInstant());
+    init();
+  }
+
+  public void init() {
+    users.stream().forEach(u -> {
+      repositories.put(u, new HashSet<>());
+
+      pullRequests.put(u, new HashSet<>());
+      openPullRequests.put(u, new HashSet<>());
+      closedPullRequests.put(u, new HashSet<>());
+
+      issues.put(u, new HashSet<>());
+      openIssues.put(u, new HashSet<>());
+      closedIssues.put(u, new HashSet<>());
+    });
+  }
+
+  public synchronized void refresh() {
+    LOGGER.info("Refreshing report data.");
+    users.stream().forEach(u -> {
+      LOGGER.info("Getting forks for user: {}.", u);
+      Set<Repository> forks = userForks(u);
+      repositories.get(u).addAll(forks);
+      LOGGER.info("User: {} forks: [{}].", u, forks.stream().map(r -> r.getName()).collect(Collectors.joining(",")));
+    });
+
+    repositories.values().stream().flatMap(s -> s.stream()).distinct().forEach(r -> {
+        openPullRequests.putAll(teamPullRequests(r, "open"));
+        pullRequests.putAll(openPullRequests);
+        closedPullRequests.putAll(teamPullRequests(r, "closed"));
+        pullRequests.putAll(closedPullRequests);
+
+        openIssues.putAll(teamIssues(r, "open"));
+        issues.putAll(openIssues);
+        closedIssues.putAll(teamIssues(r, "closed"));
+        issues.putAll(closedIssues);
+    });
+
+
+    /*
+     * Works but not the most efficient approach
+     *
+    repositories.entrySet().stream().forEach(e -> {
```

Comment:
Can this be removed? It seems like it's implemented above, no?

Reply:
I am going to remove that yeah!